src/azure-cli-core/azure/cli/core/extension/operations.py

```diff
@@ -220,10 +224,16 @@ def add_extension(cmd, source=None, extension_name=None, index_url=None, yes=Non
                 return
             logger.warning("Overriding development version of '%s' with production version.", extension_name)
         try:
-            source, ext_sha256 = resolve_from_index(extension_name, index_url=index_url)
+            source, ext_sha256 = resolve_from_index(extension_name, cur_version=version, index_url=index_url)
```

Comment:
`cur_version` refers to the version already installed if exists, please add a new argument for the specified version to be installed.

Reply:
Add `target_version` to `resolve_from_index` to find out the target version I want.